Lib/zipfile/__init__.py

```diff
@@ -444,16 +446,20 @@ def _compresslevel(self):
     def _compresslevel(self, value):
         self.compress_level = value
 
+    @property
+    def _file_mode(self):
+        hi = self.external_attr >> 16
+        if hi:
+            return stat.filemode(hi)
+
     def __repr__(self):
         result = ['<%s filename=%r' % (self.__class__.__name__, self.filename)]
         if self.compress_type != ZIP_STORED:
             result.append(' compress_type=%s' %
                           compressor_names.get(self.compress_type,
                                                self.compress_type))
-        hi = self.external_attr >> 16
+        result.append(' filemode=%r' % self.file_mode)
```

Comment:
```suggestion
        result.append(' file_mode=%r' % self.file_mode)
```
Would be more consistent with the name of the attribute. But that does change the behaviour of the repr, so I am not sure on this one.